File: leetcode/Two_Pointers/0088_Merge_Sorted_Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if (m == 0) {
        	nums1 = nums2;
        	return;
        }

        vector<int> temp (nums1.begin(), nums1.begin() + m);
        nums1 = temp;
        vector<int> temp2 (nums2.begin(), nums2.begin() + n);
        nums2 = temp2;

        int flag1 = 0, flag2 = 0;
        while (flag2 < n) {
        	if (flag1 == nums1.size()) {
        		nums1.insert(nums1.end(), nums2[flag2]);
        		flag1 ++;
        		flag2 ++;
        	}
        	else if (nums2[flag2] <= nums1[flag1]) {
        		nums1.insert(nums1.begin() + flag1, nums2[flag2]);
        		flag1 ++;
        		flag2 ++;
        	}
        	else
        		flag1 ++;
        }
    }
};

File: leetcode/Two_Pointers/0088_Merge_Sorted_Array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        for _ in range(n):
            nums1.pop()
        i, j = 0, 0
        while j < n:
            if i == len(nums1):
                nums1.append(nums2[j])
                j += 1
            elif nums1[i] <= nums2[j]:
                i += 1
            else:
                nums1.insert(i, nums2[j])
                i += 1
                j += 1
